### PR TITLE
Extend the backup.create action with password support

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Documentation for feature added with https://github.com/home-assistant/core/pull/110630

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/110630
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Updated documentation for the **Backup** integration, clarifying that all parameters for the `backup.create` action are optional.
	- Introduced a new `password` parameter for encrypting backup files.
	- Modified example YAML automation to demonstrate the use of the new password feature.

- **Documentation**
	- Enhanced clarity and usability of the backup process documentation, particularly for securing backups.

### CODE REVIEW

File: source/_integrations/backup.markdown [hunk 1]
```diff
@@ -35,7 +37,13 @@ process.
...
 to create a backup for your Home Assistant instance.
 
-The action has no additional options or parameters.
+All parameters for the `backup.create` action are optional.
+
+{% configuration_basic %}
+password:
+  description: >-
+    Encrypt the backup file with a password. When restoring a password-protected backup, you will need to provide this password. Store this password securely as there is no way to recover the backup without it.
+{% endconfiguration_basic %}
```
[the remark this answers is not in the document]
Comment (ludeeus) [reply source/_integrations/backup.markdown:46]: No, this is not the place for that.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @ludeeus :+1:

**Comment by MartinHjelmare** [comment]
We've merged the parent PR in a separate feature branch. I'll update here when we've merged the feature branch into Core dev branch.

**Comment by silamon** [comment]
@MartinHjelmare : Ready to merge? https://github.com/home-assistant/core/pull/132954

**Comment by MartinHjelmare** [comment]
We've reverted the addition of password to the service in core.

**Comment by ludeeus** [comment]
Why was that done?

**Comment by MartinHjelmare** [comment]
We didn't want to make any changes now to the service regarding password since we didn't have a clear plan how passwords should be handled. Some questions were: Should password be required for all backups, also non automatic backups? If password should be required for all backups, should we still allow a custom password in the service? Should we always use the configured password for automatic backups also for other backups? How do we migrate users that haven't set a password yet but have automations using the existing service?

**Pull request closed without merging**